src/components/MissionTable/MissionTable.js

```diff
@@ -0,0 +1,67 @@
+import Table from 'react-bootstrap/Table';
+import { Button, Stack } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+import { joinMission, leaveMission } from '../../redux/missions/missionsSlice';
+
+const MissionTable = () => {
+  const table = useSelector((state) => state.missions);
+  const dispatch = useDispatch();
+
+  const joinMissionHandler = (e) => {
+    const { target: { id } } = e;
+    dispatch(joinMission(id));
+  };
+
+  const leaveMissionHandler = (e) => {
+    const { target: { id } } = e;
+    dispatch(leaveMission(id));
+  };
+
```

Comment:
- [ ] Please note that the mission state should only be updated once in the first visit, currently it fetches the data & resets the state every time I change routes, kindly fix it 🙏 


![statebug](https://user-images.githubusercontent.com/89282221/224314987-4bc7aea2-06fd-4d95-a6d7-1ae1cb80f9fa.gif)

Reply:
@Talha-Hanif5666